npe2/manifest/schema.py

```diff
@@ -376,6 +378,84 @@ def _from_package_or_name(
 
     ValidationError = ValidationError  # for convenience of access
 
+    @classmethod
+    def _from_npe1_plugin(cls, plugin_name: str):
+        from .._from_npe1 import npe1_plugin_manager  # type: ignore
+
+        plugin_manager = npe1_plugin_manager()
+        if plugin_name not in plugin_manager.plugins:
+            # TODO: it would be nice to add some logic to prevent confusion here.
+            # for example... if the plugin name doesn't equal the package name, we
+            # should still be able to find it if the user gives a package name
+            raise ValueError(
+                f"Could not find plugin {plugin_name!r}\n"
+                f"Found {set(plugin_manager.plugins)}"
+            )
+
+        module = plugin_manager.plugins[plugin_name]
+        standard_meta = plugin_manager.get_standard_metadata(plugin_name)
+        package = standard_meta["package"].replace("-", "_")
+
+        commands = []
+        readers = []
+        writers = []
+        for caller in plugin_manager._plugin2hookcallers[module]:
+            for impl in caller.get_hookimpls():
+                if impl.plugin_name != plugin_name:
+                    continue
+                name = impl.specname.replace("napari_", "")
+                id = f"{package}.{name}"
+                py_name = f"{impl.function.__module__}:{impl.function.__qualname__}"
+                title = " ".join(name.split("_")).title()
+                commands.append({"id": id, "python_name": py_name, "title": title})
+
+                if "reader" in id:
+                    # add this to the readers list
+                    readers.append({"command": id, "accepts_directories": True})
+
+                write_layers = []
+                write_directory = False
+                if "write" in id:
+                    # add this to the writers list
+                    if "get_writer" not in id:
+                        layer = id.split("write_")[1]
+                        if layer != "directory":
+                            writers.append(
+                                {"command": id, "layer_types": [layer], "name": layer}
+                            )
+                            write_layers.append(layer + "*")
+                        else:
+                            id_dir = id
+                            write_directory = True
+                if write_directory:
+                    if len(write_layers) > 0:
+                        writers.append({"command": id_dir, "layer_types": write_layers})
+                    else:
+                        writers.append({"command": id_dir, "layer_types": ["*"]})
+
+                themes = []
+                if "theme" in id:
+                    from napari.utils.theme import get_theme
```

Comment:
don't import napari here.  see the [specification for this hookspec](https://napari.org/plugins/stable/hook_specifications.html#gui-hooks).  It's a function that returns a dict of `{theme_name: theme_dict}`.  So you can just call the function directly and add each theme:
```python
for name, theme in impl.function():
    # probably safer to cast to manifest.themes.ThemeColors here to make sure the keys match
    themes.append({"label": name, id:  "type": "dark", "colors": theme})
```